views/js/tools/getStatus.js

```diff
@@ -13,50 +13,46 @@
  *
- * Copyright (c) 2016-2017 (original work) Open Assessment Technologies SA ;
+ * Copyright (c) 2016-2021 (original work) Open Assessment Technologies SA ;
  */
-/**
- * @author Jean-Sébastien Conan <jean-sebastien@taotesting.com>
- * @author dieter <dieter@taotesting.com>
- */
-define(['lodash'], function (_) {
+define(['lodash'], function(_) {
     'use strict';
 
     /**
      * Gets the correct status message for a given percentage from a list of thresholds.
-     * @param {Number|String} percentage - The actual percentage. Must be comprised between 0 and 100.
+     * @param {number|string} percentage - The actual percentage. Must be comprised between 0 and 100.
      *                                     Other values will be adjusted to fit the interval.
-     * @param {Array|Object} [thresholds] - A list of descriptors for each thresholds.
+     * @param {Array|object} [thresholds] - A list of descriptors for each thresholds.
      *                                      A threshold field must be provided for each.
-     * @param {Object} [opts]
-     * @param {Object} [minimumGlobalPercentage] - lowest value that will be used in the global score computation
-     * @returns {Object} Returns the corresponding threshold, or an empty object if none match.
+     * @param {object} [opts]
+     * @param {object} [minimumGlobalPercentage] - lowest value that will be used in the global score computation
+     * @returns {object} Returns the corresponding threshold, or an empty object if none match.
      */
     return function getStatus(percentage, thresholds, opts) {
-        var options = opts || {};
-        var testPercentage = Math.max(0, Math.min(100, Math.round(parseInt(percentage, 10) || 0)));
-        var globalPercentage = (options.minimumGlobalPercentage)
+        const options = opts || {};
+        const testPercentage = Math.max(0, Math.min(100, Math.round(parseInt(percentage, 10) || 0)));
+        const globalPercentage = options.minimumGlobalPercentage
             ? Math.max(testPercentage, options.minimumGlobalPercentage)
             : testPercentage;
 
         // need a structure compatible with the handlebars template
-        var status = {
+        const status = {
             // the percentage is between 0 and 100 and obviously must be a number
             percentage: testPercentage,
             globalPercentage: globalPercentage,
             quality: {}
         };
-        var len, feedback, i, step;
 
         // grab a feedback related to the percentage in the thresholds list
         if (thresholds) {
-            if (!_.isArray(thresholds)) {
+            if (!Array.isArray(thresholds)) {
                 thresholds = [thresholds];
             }
 
-            len = thresholds.length;
-            for (i = 0; i < len; i++) {
-                step = thresholds[i];
+            let feedback;
+            const len = thresholds.length >>> 0;
```

Comment:
Is this ok? It is first time I see `>>>`

Reply:
Local optimisation often used for loop over arrays, forcing the number to become an `Int32`.
You can see an example here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#polyfill